platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/configurationcache/serialization/codecs/TaskNodeCodec.kt

```diff
@@ -306,7 +305,7 @@ suspend fun WriteContext.writeRegisteredPropertiesOf(
                 }
 
                 is RegisteredProperty.Input -> {
-                    val finalValue = InputParameterUtils.prepareInputParameterValue(propertyValue)
+                    val finalValue = DeferredUtil.unpackNestableDeferred(propertyValue)
```

Comment:
Unpack deferreds but do not resolve providers/factories, nor coherce `GString` into `String`

Reply:
Yes, coercion from `GString` to `String` should happen at a different level uniformly for all properties.